README.md

```diff
@@ -13,58 +13,81 @@ npm i @itwin/eslint-plugin --save-dev
 
 ## Usage
 
-Add `@itwin` to the plugins section of your eslint configuration and (optionally) extend one of the provided configs. You can omit the `/eslint-plugin` suffix:
+Create an `eslint.config.js` file at the root of your project. Inside this file, import any of the provided configs and add it to the exported modules:
 
-```json
-{
-  "plugins": ["@itwin"],
-  "extends": "plugin:@itwin/itwinjs-recommended"
-}
+```javascript
+const itwinjsRecommended = require("@itwin/eslint-plugin/dist/configs/itwinjs-recommended");
+const itwinJsdoc = require("@itwin/eslint-plugin/dist/configs/jsdoc");
+
+module.exports = [
+  itwinjsRecommended,
+  itwinJsdoc,
+];
 ```
 
-Then configure the rules you want to override under the rules section.
+Then configure the rules you want to override by adding a section with which files to apply the rule overrides to.
 
-```json
-{
-  "rules": {
-    "@itwin/rule-name": "off"
+```javascript
+const itwinjsRecommended = require("@itwin/eslint-plugin/dist/configs/itwinjs-recommended");
+const itwinJsdoc = require("@itwin/eslint-plugin/dist/configs/jsdoc");
+
+module.exports = [
+  itwinjsRecommended,
+  itwinJsdoc,
+  {
+    files: ["**/*.{ts,tsx}"],
+    rules: {
+      "@typescript-eslint/no-explicit-any": "error",
+    }
   }
-}
+];
 ```
 
 ## Using with VSCode
 
-VSCode has an ESLint plugin, but it has some issues with plugin resolution. In order to use this config without errors, it needs to be configured to resolve plugins relative to this package (in `.vscode/settings.json`):
+In order for VSCode to use the config file as it is set up, add the following setting to the the VSCode settings (in `.vscode/settings.json`):
 
 ```json
-"eslint.options": {
-  "resolvePluginsRelativeTo": "./node_modules/@itwin/eslint-plugin",
-  ...
-},
+"eslint.experimental.useFlatConfig": true,
 ```
 
-As a side effect, any additional plugins added in consumer packages won't be loaded. If you want to use another ESLint plugin, there are two options:
-
-1. Submit a PR to add the ESLint plugin (and an accompanying optional configuration) to this package.
-2. Add all the plugins used in this package along with the new one to your package's devDependencies and remove the above configuration from settings.json.
-
 ## Rules not in recommended configs
 
 ### `no-internal` - prevents use of internal/alpha APIs. Example configurations
 
-```json
+```javascript
 // custom config
-"@itwin/no-internal": [
-"error",
+const itwinjsRecommended = require("@itwin/eslint-plugin/dist/configs/itwinjs-recommended");
+const noInternal = require("@itwin/eslint-plugin/dist/rules/no-internal");
+
+module.exports = [
+  itwinjsRecommended,
   {
-    "tag": ["internal", "alpha", "beta"]
+    files: ["src/**/*.{ts,tsx}"],
+    plugins: {
```

Comment:
I'm not a fan of defining a shim plugin here, shouldn't the user be able to just import this package as a plugin and then refer to its rules directly?

Reply:
new way to do this added and works